src/components/avatar/Avatar.tsx

```diff
@@ -2,67 +2,61 @@ import * as React from 'react';
 import classNames from 'classnames';
 import { Text, Icon } from '@/index';
 import { BaseProps, extractBaseProps } from '@/utils/types';
-// import { TooltipProps } from '@/index.type';
 import { AccentAppearance } from '@/common.type';
 
 export type AvatarSize = 'regular' | 'tiny';
-
 export interface AvatarProps extends BaseProps {
   /**
    * Color of the `Avatar`
    */
-  appearance?: AccentAppearance;
-  /**
-   * **Only first 2 characters are rendered (SOON TO BE DEPRECATED)**
-   */
-  children?: string;
+  appearance: AccentAppearance;
   /**
-   * First Name
+   * Text to show in `Avatar`
    */
-  firstName?: string;
+  text?: string;
   /**
-   * Last Name
+   * Icon to show in `Avatar`
    */
-  lastName?: string;
-  /**
-   * Determines if tooltip is visible
-   */
-  withTooltip: boolean;
+  icon?: string;
+
+  image?: {
+    /**
+     * Image url to show in `Avatar`
+     */
+    src: string;
+    /**
+     * Image alt text to show in `Avatar`
+     */
+    altText: string;
+  };
   /**
-   * Position to place the tooltip
+   * Tooltip text to show in `Avatar`
    */
-  // tooltipPosition: TooltipProps['position'];
+  tooltip?: string;
   /**
-   * Determines size of `Avatar`
+   * Tooltip text to show in `Avatar`
    */
   size: AvatarSize;
 }
 
-const initialsLength = 2;
-
 export const Avatar = (props: AvatarProps) => {
-  const { withTooltip, size, children, firstName, lastName, className, appearance } = props;
+  const { size, className, appearance, icon, text, tooltip, image, ...rest } = props;
 
   const baseProps = extractBaseProps(props);
 
-  const initials = children
-    ? children.trim().slice(0, initialsLength)
-    : `${firstName ? firstName.trim()[0] : ''}${lastName ? lastName.trim()[0] : ''}`;
-
-  // const tooltip = children || `${firstName || ''} ${lastName || ''}` || '';
   const DefaultAppearance = 'secondary';
 
   const colors = ['accent4', 'primary', 'accent3', 'alert', 'accent2', 'warning', 'accent1', 'success'];
 
-  const AvatarAppearance =
-    appearance || colors[(initials.charCodeAt(0) + (initials.charCodeAt(1) || 0)) % 8] || DefaultAppearance;
+  const AvatarAppearance = appearance || colors[Math.floor(Math.random() * colors.length)] || DefaultAppearance;
```

Comment:
According to this condition, **AvatarAppearance** will always be **primary**

Reply:
Remove from default prop